src/main/java/cinema/dao/impl/UserDaoImpl.java

```diff
@@ -29,15 +29,10 @@ public User add(User user) {
                 transaction.rollback();
             }
             throw new DataProcessingException("Can't create user : " + user, e);
-        } finally {
-            if (session != null) {
-                session.close();
-            }
         }
         return user;
     }
 
-    @Override
     public User findByEmail(String email) {
```

Comment:
somewhere you are using name findBySmth, in another place getBySmth. Let's choose one variant and stick to it.